### PR TITLE
Update hybrid jobs demos

### PR DESCRIPTION
**Pull request by mudit2812**
The hybrid jobs were failing. Some of them were because of incorrect variable names. Others because functions defined outside the hybrid job context were being used inside it. This PR fixes both of those.

From discussions outside GitHub:

> * The `getting_started_with_hybrid_jobs` demo uses some pickling and is only expected to work with 0.33.1. Once changes in this PR are incorporated, it does work with 0.33.1. An empty dictionary for results is returned with the most recent PL, and it's not a bug, its built in on the Braket end.
> * Once the fixes for `braket_parallel_gradients` are added, everything runs, the results are just incorrect. It's a static demo, so it will continue to look like the results are correct unless someone actually tries it. We'll let AWS know.

There is also a deprecation warning about `Observable.return_types` in the `braket_parallel_gradients` demo, but it's originating from the braket plugin. Their release is incosistent with ours, so it will be fixed, but probably not in time for 0.34

### CODE REVIEW

File: demonstrations/getting_started_with_hybrid_jobs.py [hunk 1]
```diff
@@ -232,7 +250,11 @@ def qubit_rotation_hybrid_job(num_steps=1, stepsize=0.5):
 import pandas as pd
 import matplotlib.pyplot as plt
 
-df = pd.DataFrame(job.metrics())
+metrics = job.metrics()
+print(metrics)
+
+df = pd.DataFrame(metrics)
+print(df)
```
Comment (timmysilv) [inline demonstrations/getting_started_with_hybrid_jobs.py:257]: should these `print` statements be here, and if so, should the outputs be added to the demo outputs somewhere?

File: demonstrations/getting_started_with_hybrid_jobs.py [hunk 2]
```diff
@@ -169,7 +169,25 @@ def qubit_rotation(num_steps=10, stepsize=0.5):
 
 @hybrid_job(device="local:pennylane/lightning.qubit")
 def qubit_rotation_hybrid_job(num_steps=1, stepsize=0.5):
-    return qubit_rotation(num_steps=num_steps, stepsize=stepsize)
```
Comment (timmysilv) [inline demonstrations/getting_started_with_hybrid_jobs.py:172]: if we aren't using this anymore, we should update the end of the text blurb above

File: demonstrations/getting_started_with_hybrid_jobs.py (hunk 1 above)
Comment (mudit2812) [reply demonstrations/getting_started_with_hybrid_jobs.py:257]: I thought I removed those 😅

**Review by timmysilv — APPROVED** [review]

**Review by KetpuntoG — COMMENTED** [review]
Thanks! Should we update the warning on the demo to PL 0.34?

**Comment by lillian542** [comment]
> Thanks! Should we update the warning on the demo to PL 0.34?

It's still compatible with 0.32, as claimed in the warning, and when we were testing, it also worked with 0.33 and 0.33.1. It is **not** compatible with 0.34, because of things defined on the Braket end. So I think we should leave the warning as-is, or maybe say "only guaranteed to be compatible with Python 3.10 and PennyLane 0.32"

**Review by KetpuntoG — APPROVED** [review]
LGTM thanks!

**Pull request merged**